src/transformers/models/vits/modeling_vits.py

```diff
@@ -1487,3 +1766,420 @@ def forward(
+
+        if labels_attention_mask is not None:
+            labels_padding_mask = labels_attention_mask.unsqueeze(1).float()
+        else:
+            labels_attention_mask = torch.ones((labels.shape[0], labels.shape[2])).float().to(self.device)
+            labels_padding_mask = labels_attention_mask.unsqueeze(1)
+
+        text_encoder_output = self.text_encoder(
+            input_ids=input_ids,
+            padding_mask=input_padding_mask,
+            attention_mask=attention_mask,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+        hidden_states = text_encoder_output[0] if not return_dict else text_encoder_output.last_hidden_state
+        hidden_states = hidden_states.transpose(1, 2)
+        input_padding_mask = input_padding_mask.transpose(1, 2)
+        prior_means = text_encoder_output[1] if not return_dict else text_encoder_output.prior_means
+        prior_log_variances = text_encoder_output[2] if not return_dict else text_encoder_output.prior_log_variances
+
+        latents, posterior_means, posterior_log_variances = self.posterior_encoder(
+            labels, labels_padding_mask, speaker_embeddings
+        )
+        prior_latents = self.flow(latents, labels_padding_mask, speaker_embeddings, reverse=False)
+
+        prior_means, prior_log_variances = prior_means.transpose(1, 2), prior_log_variances.transpose(1, 2)
+        with torch.no_grad():
+            # negative cross-entropy
+
+            # [batch_size, d, latent_length]
+            prior_variances = torch.exp(-2 * prior_log_variances)
+            # [batch_size, 1, latent_length]
+            neg_cent1 = torch.sum(-0.5 * math.log(2 * math.pi) - prior_log_variances, [1], keepdim=True)
+            # [batch_size, text_length, d] x [batch_size, d, latent_length] = [batch_size, text_length, latent_length]
+            neg_cent2 = torch.matmul(-0.5 * (prior_latents**2).transpose(1, 2), prior_variances)
+            # [batch_size, text_length, d] x [batch_size, d, latent_length] = [batch_size, text_length, latent_length]
+            neg_cent3 = torch.matmul(prior_latents.transpose(1, 2), (prior_means * prior_variances))
+            # [batch_size, 1, latent_length]
+            neg_cent4 = torch.sum(-0.5 * (prior_means**2) * prior_variances, [1], keepdim=True)
+
+            # [batch_size, text_length, latent_length]
+            neg_cent = neg_cent1 + neg_cent2 + neg_cent3 + neg_cent4
+
+            attn_mask = torch.unsqueeze(input_padding_mask, 2) * torch.unsqueeze(labels_padding_mask, -1)
+
+            attn = monotic_alignment_function(neg_cent, attn_mask.squeeze(1)).unsqueeze(1).detach()
+
+        durations = attn.sum(2)
+
+        if self.config.use_stochastic_duration_prediction:
+            log_duration = self.duration_predictor(
+                hidden_states, input_padding_mask, speaker_embeddings, durations=durations, reverse=False
```

Comment:
How come self.noise_scale_duration isn't passed in here?

Reply:
It's because it's in training mode (i.e `reverse=False`). In that case the duration predictor doesn't use `noise_scale`